### PR TITLE
SeriesScanUtil support predicate pushdown

### PR DESCRIPTION
**Pull request by liuminghui233**
See https://apache-iotdb.feishu.cn/docx/HCEqdNfdtoW6B1xlpjOcRMwsnce#RPMVdYu2yoK9jYxz4Ecca6yHnkg.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java [hunk 1]
```diff
@@ -1192,8 +1221,8 @@ TsBlock getAllSatisfiedPageData(boolean ascending) throws IOException {
       }
     }
 
-    void setFilter(Filter filter) {
-      data.setFilter(filter);
+    void setPushDownFilter(Filter pushDownFilter) {
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java:1224]:
```suggestion
    void addPushDownFilter(Filter pushDownFilter) {
```

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java [hunk 2]
```diff
@@ -1111,26 +1148,29 @@ private void unpackUnseqTsFileResource() throws IOException {
...
   }
 
   protected IPointReader getPointReader(TsBlock tsBlock) {
     return tsBlock.getTsBlockSingleColumnIterator();
   }
 
-  Filter getGlobalTimeFilter() {
-    return scanOptions.getGlobalTimeFilter();
+  protected boolean timeAllSelected(IMetadata metadata) {
+    return metadata.timeAllSelected();
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java:1164]: fixed

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java [hunk 3]
```diff
@@ -1097,12 +1139,7 @@ private void unpackSeqTsFileResource() throws IOException {
...
     ITimeSeriesMetadata timeseriesMetadata =
-        loadTimeSeriesMetadata(
-            orderUtils.getNextUnseqFileResource(true),
-            seriesPath,
-            context,
-            getGlobalTimeFilter(),
-            scanOptions.getAllSensors());
+        loadTimeSeriesMetadata(orderUtils.getNextUnseqFileResource(true), seriesPath, context);
     if (timeseriesMetadata != null) {
       timeseriesMetadata.setModified(true);
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java:1139]: fixed

File: iotdb-core/tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/block/TsBlock.java [hunk 4]
```diff
@@ -223,6 +223,18 @@ public Column[] getColumns(int[] columnIndexes) {
...
+    for (int i = 0; i < valueColumns.length; i++) {
+      if (valueColumns[i].isNull(rowIndex)) {
+        rowValues[i] = null;
+      } else {
+        rowValues[i] = valueColumns[i].getObject(rowIndex);
+      }
+    }
+    return rowValues;
+  }
+
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply iotdb-core/tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/block/TsBlock.java:237]: removed

File: iotdb-core/tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/operator/Not.java [hunk 5]
```diff
@@ -57,6 +58,11 @@ public boolean satisfyRow(long time, Object[] values) {
     return !filter.satisfyRow(time, values);
   }
 
+  @Override
+  public boolean[] satisfyTsBlock(TsBlock tsBlock) {
+    return filter.satisfyTsBlock(tsBlock);
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply iotdb-core/tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/operator/Not.java:63]: fixed

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java (hunk 1 above)
Comment (liuminghui233) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/SeriesScanUtil.java:1224]: fixed

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**